kiwi_lint/function_based_views.py

```diff
@@ -0,0 +1,133 @@
+"""function_based_views.py provides a pylint checker FunctionBasedViewChecker that warns
+    against using function based views in Django.
+"""
+
+from copy import deepcopy
+from importlib import import_module
+
+import astroid
+
+import django
+from django.apps import apps as django_apps
+from django.conf import settings
+from django.urls.resolvers import URLResolver, URLPattern
+
+from pylint import checkers
+from pylint import interfaces
+
+
+class DjangoViewsVisiter(checkers.BaseChecker):
+    """Base class for visiting only astroid modules which contain django views.
+
+    Derivatives could override `visit_views_module` and/or `leave_views_module` to hook just into
+    the walking of view containing astroid modules.
+
+    Class Attributes:
+
+    Instance Attributes:
+        url_mapping
+        Type :: dict[str, 2-tuple(module_name: str, view_name: str)]
+
+        installed_apps_function_filters
+        Type :: iterable[callable[all_apps: iterable[str]] -> iterable[str]]
+
+            List of callbacks, which will be called for pruning the installed apps,
+            on whose views the checker should be invoked.
+    """
+
+    def __init__(self, *args, main_urls_package=None, filters=None, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.installed_apps_function_filters = filters or []
+        django.setup()
+
+        self.__installed_apps = self._resolve_app_configs(self._get_django_project_apps())
+
+        project_urls = import_module(main_urls_package)
```

Comment:
This is always `settings.ROOT_URLCONF` (points to the urls module) so you don't need to pass this externally as parameter.